src/python/pants/base/specs.py

```diff
@@ -192,6 +192,24 @@ def make_glob_patterns(self, address_mapper):
     ]
 
 
+_specificity = {
+  SingleAddress: 0,
+  SiblingAddresses: 1,
+  AscendantAddresses: 2,
```

Comment:
Is it possible for an end-user to type an ascendant address through the CLI, or is this only used in internal code? I'm wondering if this should even belong here.

Also, what is the reasoning for an ascendant address being more specific than a descendant address? I'm not able to give a good technical reason for this, but a descendant address "feels" more specific intuitively to me.

Reply:
I don't really know what ascendant addresses are for. I just know that they're there, so I had to do something with them. To me they are less specific because they span multiple levels, whereas siblings are all in the same dir.

Reply:
Does it fail if we remove them from the dictionary? I suppose we should regardless have it as a matter of defensive programming.

Agreed that ascendant is less specific than sibling. I mean that ascendant also seems less specific than descendent. I would order it:

single > sibling > descendant > ascendant